src/wallet/wallet.cpp

```diff
@@ -4141,7 +4144,7 @@ CWallet* CWallet::CreateWalletFromFile(const std::string& name, const fs::path&
     }
 
     walletInstance->m_last_block_processed = chainActive.Tip();
-    RegisterValidationInterface(walletInstance);
+    RegisterValidationInterface(temp_wallet.release());
```

Comment:
commit: [wallet] Fix potential memory leak in CreateWalletFromFile

This isn't a complete fix because there are still early returns with errors in the rescan logic below. I believe it is safe to move the `RegisterValidationInterface` call to the bottom of the function as I argued in #12647 because `cs_main` gets locked on line 4135.

Reply:
Done